fs/fs_zenfs.cc

```diff
@@ -920,6 +1001,36 @@ Status ZenFS::DecodeFileDeletionFrom(Slice* input) {
   return Status::OK();
 }
 
+Status ZenFS::DecodeFileLinkDeletionFrom(Slice* input) {
+  uint64_t fileID;
+  uint32_t nrLinks;
+  std::string fileName;
+  Slice slice;
+
+  if (!GetFixed64(input, &fileID))
+    return Status::Corruption("Zone file link deletion: file id missing");
+
+  if (!GetFixed32(input, &nrLinks))
+    return Status::Corruption("Zone file link deletion: link count mismatch");
+
+  if (!GetLengthPrefixedSlice(input, &slice))
+    return Status::Corruption("Zone file link deletion: file name missing");
+
+  fileName = slice.ToString();
+  if (files_.find(fileName) == files_.end())
+    return Status::Corruption("Zone file link deletion: no such file");
+
+  std::shared_ptr<ZoneFile> zoneFile = files_[fileName];
+  if (zoneFile->GetID() != fileID)
+    return Status::Corruption("Zone file link deletion: file ID missmatch");
+
+  zoneFile->RemoveLinkInfo(fileName);
+  files_.erase(fileName);
+  zoneFile->SetNrLinks(nrLinks);
```

Comment:
I would say `RemoveLinkInfo` should update the `nr_links_` correctly and we should not overwrite that value with `SetNrLinks(nrLinks)`. We could check at this point `if (nrLinks == zoneFile->GetNrLinks())`.

Reply:
You are right. SetNrLinks() is redundant at this point, will remove it.